crates/network/src/config.rs

```diff
@@ -1,8 +1,13 @@
-use std::{fmt, time};
+use core::fmt::Formatter;
+use core::time::Duration;
```

Comment:
Where possible, we should ideally look to import from `core` instead of `std`, which paves the way for `no_std` compatibility, which is something we should bear in mind given we are targeting Wasm (smaller builds).

https://rust-lang.github.io/rust-clippy/master/index.html#std_instead_of_core

Reply:
Idk, right now, it's not very valuable to do this. Seeing as these crates still use `std` unconditionally, there's no functional reason to do it at this time without introducing unnecessary confusion, WASM support will be investigated in a future effort, at which point (armed with functional purpose) we can focus on details like these.

Here in `network::discovery::state`, we're dependent on:

https://github.com/calimero-network/core/blob/db771010bbabbdf81dfc4fa5e412eebce4ae410e/crates/network/src/discovery/state.rs#L1-L8

It's also not all crates that will undergo this effort, parts like the server, runtime, rocks store etc won't be compiled to WASM.

Reply:
It's something that I've gotten into the habit of doing over the years. It's becoming increasingly common practice, and is a Rust recommendation. It causes no harm.